sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/Extensions/ResourceManagementClientExtensions.cs

```diff
@@ -0,0 +1,67 @@
+// ------------------------------------------------------------------------------------------------
+// <copyright file="ResourceManagementClientExtensions.cs" company="Microsoft Corporation">
+//   Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------------------------
+
+namespace DnsResolver.Tests.Extensions
+{
+    using System;
+    using System.Linq;
+    using Microsoft.Azure.Management.Resources;
+    using Microsoft.Azure.Management.Resources.Models;
+
+    internal static class ResourceManagementClientExtensions
+    {
+        public static ResourceGroup CreateResourceGroup(
+            this ResourceManagementClient client,
+            string resourceGroupName = null,
+            string resourceGroupLocation = null)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            resourceGroupName = resourceGroupName ?? TestDataGenerator.GenerateResourceGroupName();
+            var response = client.ResourceGroups.CreateOrUpdate(resourceGroupName, TestDataGenerator.GenerateResourceGroup(resourceGroupLocation));
+
+            return response;
+        }
+
+        public static string GetResourceLocation(this ResourceManagementClient client, string resourceType)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            string location = null;
+
+            var parts = resourceType.Split('/');
+            var providerName = parts[0];
```

Comment:
Do we not want some validation on `resourceType` being the right format? Else, we risk a NPE here.